blockchains/xrp/xrp_worker.js

```diff
@@ -51,23 +53,33 @@ class XRPWorker extends BaseWorker {
     this.lastConfirmedBlock = parseInt(lastValidatedLedgerData.ledger.ledger_index) - constants.CONFIRMATIONS;
   }
 
-  async fetchLedgerTransactions(connection, ledger_index) {
-    const ledger = await this.connectionSend(connection, {
+  async fetchLedger(connection, ledger_index, should_expand) {
+    const result = await this.connectionSend(connection, {
       command: 'ledger',
       ledger_index: parseInt(ledger_index),
       transactions: true,
-      expand: false
-    }).then(value => value.result.ledger);
+      expand: should_expand
+    });
 
-    assert(ledger.closed === true);
+    const ledger = result.result.ledger;
 
-    if (typeof ledger.transactions === 'undefined' || ledger.transactions.length === 0) {
-      return { ledger: ledger, transactions: [] };
+    assert(ledger.closed === true);
+    assert(typeof ledger.transactions !== 'undefined');
+    assert(typeof ledger.transaction_hash !== 'undefined');
+
+    if (ledger.transactions.length === 0 && ledger.transaction_hash !== this.emptyTransactionHash) {
+      // This block is invalid, the problem must be in the Endpoint. Wait and retry.
+      await new Promise((resolve) => setTimeout(resolve, this.retryIntervalMs));
+      logger.info(`Ledger ${ledger_index} is being retried due to invalid response`);
+      return await this.fetchLedger(connection, ledger_index, should_expand);
```

Comment:
I know that in theory, k8s would kill the pod if it takes too much time, but isn't it a bad idea to leave an open recursion like this? Wouldn't a function with a retry mechanism be better? If it fails X times, then adios?

Reply:
retry looks better, agree